packages/frontend/src/components/DashboardFilter/ActiveFilters/index.tsx

```diff
@@ -121,6 +130,74 @@ const ActiveFilters: FC<ActiveFiltersProps> = ({
     });
     const dragSensors = useSensors(mouseSensor, touchSensor);
 
+    const sortedTabUuids = useMemo(() => {
+        const sortedTabs = dashboardTabs?.sort((a, b) => a.order - b.order);
+        return sortedTabs?.map((tab) => tab.uuid) || [];
+    }, [dashboardTabs]);
+
+    const makeTabGetter = useCallback(
+        (filters: DashboardFilters) => {
+            if (!dashboardTiles) return () => [];
+            return (filterId: string) => {
+                const { tabsUsingThisFilter, otherTabsWithFilterableTiles } =
+                    dashboardTiles.reduce(
+                        (acc, tile) => {
+                            if (!tile.tabUuid || !isTileFilterable(tile))
+                                return acc;
+
+                            const filterIdsForTile =
+                                getDashboardFilterRulesForTile(
+                                    tile.uuid,
+                                    filters.dimensions,
+                                ).map((tileFilter) => tileFilter.id);
+
+                            if (filterIdsForTile.includes(filterId)) {
+                                if (
+                                    !acc.tabsUsingThisFilter.includes(
+                                        tile.tabUuid,
+                                    )
+                                ) {
+                                    acc.tabsUsingThisFilter.push(tile.tabUuid);
+                                }
+                            } else if (
+                                !acc.otherTabsWithFilterableTiles.includes(
+                                    tile.tabUuid,
+                                )
+                            ) {
+                                acc.otherTabsWithFilterableTiles.push(
+                                    tile.tabUuid,
+                                );
+                            }
+                            return acc;
+                        },
+                        {
+                            tabsUsingThisFilter: [] as string[],
+                            otherTabsWithFilterableTiles: [] as string[],
+                        },
+                    );
+
+                return sortedTabUuids.filter((tabUuid: string) => {
+                    return (
+                        tabsUsingThisFilter.includes(tabUuid) ||
+                        !otherTabsWithFilterableTiles.includes(tabUuid)
+                    );
+                });
+            };
+        },
+        [dashboardTiles, sortedTabUuids],
+    );
+
+    const getTabsUsingFilter = useCallback(
+        (filterId: string) => makeTabGetter(dashboardFilters)(filterId),
+        [makeTabGetter, dashboardFilters],
```

Comment:
minor: I think wrapping getTabsUsing filter and other tab getters within the useCallback is redundant here, it's used within the same react component and gets called on every render